view/frontend/web/js/product/options.js

```diff
@@ -112,18 +128,14 @@ define([
                 });
                 validSuperAttribute =  Boolean(validSuperAttribute && validSuperAttribute.search("false") === -1);
 
-                var isValid = $(this).valid() && validSuperAttribute;
+                document.customOptions = document.getSkuForOptions(selectedOptions);
+                document.resetPriceAndSku(true);
 
             }else{
-                var isValid = $(this).valid();
-            }
 
-            if (isValid) {
                 document.customOptions = document.getSkuForOptions(selectedOptions);
-                $('.button-prevent').hide();
                 document.resetPriceAndSku(true);
```

Comment:
why are we calling the same code at the end of the if block as the else block? also it looks like the true being passed in refers to the item being configurable, but in the else block, that should mean the item isn't configurable, so that should be false right? Can you look into what this code is meant to be doing, and if everything is doing what it should here?

Reply:
You are right, I missed that one, will correct